docs/decisions/0011-producing-to-event-bus-via-settings.rst

```diff
@@ -0,0 +1,92 @@
+11. Enable producing to event bus via settings
+##############################################
+
+Status
+******
+
+**Accepted** 2023-06-05
+
+Context
+*******
+
+The initial implementation of the event bus allowed only a single event type to be published to a topic, with details like topic name, consumer group name, and consumer name were configured via code. We weren't sure what the api would finally look like, and whether the event bus config would live as part of the definition of the event, so we just went with explicit code that we could iterate upon.
+
+The current implementation of openedx-events does not actually push any events to the underlying implementations like `edx-event-bus-kafka`_ and `edx-event-bus-redis`_. The event-producing application is expected to create a signal handler (since openedx-events subclasses Django signals) to catch the event and push it into the event bus. Some examples of the handlers: `handlers example`_.
+
+.. _handlers example: https://github.com/openedx/edx-platform/blob/27b8d2f68d5dfaf84755e7d7f8dccc97ce3be509/cms/djangoapps/contentstore/signals/handlers.py#L162-L210
+.. _edx-event-bus-kafka: https://github.com/openedx/event-bus-kafka
+.. _edx-event-bus-redis: https://github.com/openedx/event-bus-redis
+
+This ADR aims to propose a solution for configuring the details, like the topic name, consumer group name, etc. via Django settings as well as pushing events to the event bus without requiring the producing application to write additional handlers.
+
+
+Decision
+********
+
+Update constructor of ``OpenEdxPublicSignal`` class to optionally accept ``topics`` and ``event_key_field`` and store them. These values can be then used as default while pushing them to event bus implementations. Add ``topics`` and ``event_key_field`` for existing signal definitions in openedx_events repository. Once this ADR is implemented, make ``topics`` and ``event_key_field`` required in ``OpenEdxPublicSignal`` constructor.
```

Comment:
I'd love to also get the thoughts of others, but I don't think I'd push this all the way to the event definition. I was thinking about the defaults simply being in the `EVENT_BUS_PRODUCER_CONFIG` that is defined in each service's repository. Thoughts?

Reply:
Makes sense. We could add currently used events in the defaults. For example, add [these events](https://github.com/openedx/edx-platform/blob/27b8d2f68d5dfaf84755e7d7f8dccc97ce3be509/cms/djangoapps/contentstore/signals/handlers.py#L162-L210) for edx-platform.